src/sql-parser/src/parser.rs

```diff
@@ -7199,6 +7202,45 @@ impl<'a> Parser<'a> {
             query,
         }))
     }
+    /// Parse an `EXPLAIN [KEY|VALUE] SCHEMA` statement assuming that the `EXPLAIN [KEY|VALUE]` tokens
+    /// have already been consumed
+    fn parse_explain_schema(&mut self) -> Result<Statement<Raw>, ParserError> {
+        self.prev_token();
+        let schema_for = match self.expect_one_of_keywords(&[KEY, VALUE])? {
+            KEY => ExplainSchemaFor::Key,
+            VALUE => ExplainSchemaFor::Value,
+            _ => unreachable!(),
+        };
+
+        self.expect_keyword(SCHEMA)?;
+
+        let format = if self.parse_keyword(AS) {
+            match self.parse_one_of_keywords(&[JSON]) {
+                Some(JSON) => ExplainFormat::Json,
+                None => {
+                    return Err(ParserError::new(
+                        self.index,
+                        "only json format is supported",
+                    ))
+                }
+                _ => unreachable!(),
+            }
+        } else {
+            ExplainFormat::Json
+        };
+
+        self.expect_keywords(&[FOR, CREATE])?;
+
+        if let Statement::CreateSink(statement) = self.parse_create_sink()? {
```

Comment:
Currently, this is doing the planning for the create sink statement simplistically . So it's going to do all the associated checks. Maybe we can pass a flag here to do the bare minimum, thoughts?

Reply:
What's the concern with doing all the checks in `parse_create_sink`? I think it's fine as is.